src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -672,34 +651,24 @@ private List<AzureTenant> ListAccountTenants(AzureAccount account, AzureEnvironm
             }
 
             SubscriptionClient subscriptionClient = null;
-
             try
             {
+                List<AzureSubscription> subscriptionList = new List<AzureSubscription>();
                 subscriptionClient = AzureSession.ClientFactory.CreateCustomArmClient<SubscriptionClient>(
                         environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
                         new TokenCredentials(accessToken.AccessToken) as ServiceClientCredentials,
                         AzureSession.ClientFactory.GetCustomHandlers());
 
-                IPage<Subscription> subscriptions = null;
-                if (listNextLink == null)
-                {
-                    subscriptions = subscriptionClient.Subscriptions.List();
-                }
-                else
-                {
-                    subscriptions = subscriptionClient.Subscriptions.ListNext(listNextLink);
-                }
-                if (subscriptions != null)
+                AzureContext context = new AzureContext(_profile.Context.Subscription, account, environment, 
+                                            CreateTenantFromString(tenantId, accessToken.TenantId));
+
+                IEnumerable<AzureSubscription> enumerable = subscriptionClient.ListAll().Select(s => s.ToAzureSubscription(context));
+                foreach (var subscription in enumerable)
```

Comment:
Why not just return this, why do we need to return a list when the method signature is just returning an enumerable?